internal/process/queue.go

```diff
@@ -139,35 +143,43 @@ func (q *Queue) worker(queue workqueue.RateLimitingInterface, process func(key s
 	}
 }
 
-func (q *Queue) logAndUpdateWorkerTimes(key string, name string, log *slog.Logger) {
-	// log time
-	now := time.Now()
-	lastTime, ok := q.workerExecutionTimes[name]
-	if ok {
-		log.Info(fmt.Sprintf("execution - worker %s last execution time %s, executed after %s", name, lastTime, now.Sub(lastTime)))
+func (q *Queue) updateWorkerTime(key string, workerNameId string, log *slog.Logger) {
+	q.workerExecutionTimes[workerNameId] = time.Now()
+	q.workerLastKeys[workerNameId] = key
+	log.Info(fmt.Sprintf("updating worker time, processing item %s, queue length is %d", key, q.queue.Len()))
+}
+
+func (q *Queue) removeTimeIfInWarnMargin(key string, workerNameId string, log *slog.Logger) {
+
+	processingStart, ok := q.workerExecutionTimes[workerNameId]
+
+	if !ok {
+		log.Warn(fmt.Sprintf("worker %s has no start time", workerNameId))
+		return
+	}
+
+	processingTime := time.Since(processingStart)
+
+	if processingTime > q.warnAfterTime {
```

Comment:
Will we log here information when the worker is idle?
In case there is nothing in the queue to process for a long time so the last execution time (in `q.workerExecutionTimes[workerNameId]` is very old.

Reply:
In the case the time from `workerExecutionTimes` is removed in https://github.com/kyma-project/kyma-environment-broker/pull/1604/files#diff-e94ea6ee51e7a52f24c56388c5828851ee956a6cb5d30cf83a6a82cef2f7784cR119 before a chance to log it hence not producing idle logs.

Reply:
Iddle logs were produced in the healthcheck go routine.